clodius/cli/aggregate.py

```diff
@@ -754,27 +841,159 @@ def line_to_np_array(line):
+
+        interval_inserts.clear()
+        tile_inserts.clear()
+
+    for interval_idx, interval in enumerate(sorted_intervals):
+        curr_zoom = 0
+        inserted = False
+        # go through each interval from most important to least
+        while curr_zoom <= max_zoom:
+            # try to place it in the highest zoom level and go down from there
+            tile_width = tile_size * 2 ** (max_zoom - curr_zoom)
+
+            curr_pos = interval[0]
+            space_available = True
+
+            # if we have not yet inserted the interval somewhere
+            if not inserted:
+                # check if there's space at this zoom level
+                while curr_pos < interval[1]:
+                    curr_tile = math.floor(curr_pos / tile_width)
+                    tile_id = f"{curr_zoom}.{curr_tile}"
+
+                    # if any of the overlapping tiles is already filled up,
+                    # lets go to the next zoom level by setting `space_available`
+                    # to false
+                    if tile_counts[tile_id] >= max_per_tile:
+                        space_available = False
+                        break
+
+                    curr_pos += tile_width
+
+            # if there is, then increment the tile counters
+            if not inserted and space_available:
+                curr_pos = interval[0]
+                while curr_pos < interval[1]:
+                    curr_tile = math.floor(curr_pos / tile_width)
+                    tile_id = "{}.{}".format(curr_zoom, curr_tile)
+
+                    tile_counts[tile_id] += 1
+
+                    curr_pos += tile_width
+
+            if inserted or space_available:
+                # there's available space
+                if not inserted:
+                    value = uid_to_interval[interval[-1]]
+                    interval_inserts.append(
+                        (
+                            interval_idx,
+                            curr_zoom,
+                            value["importance"],
+                            value["startPos"],
+                            value["endPos"],
+                            value["chrOffset"],
+                            value["uid"],
+                            value["name"],
+                            value["fields"],
+                        )
+                    )
+                    if verbose and interval_idx == 0:
+                        print(f"Interval 0 first appears at zoom level {curr_zoom}")
+
+                curr_pos = interval[0]
+                while curr_pos < interval[1]:
+                    curr_tile_x = math.floor(curr_pos / tile_width)
+                    tile_idx = tiles_cumsum[curr_zoom] + curr_tile_x
+
+                    tile_inserts.append((tile_idx, interval_idx))
+
+                    if verbose and interval_idx == 0:
+                        print(f"Interval 0 is added to {curr_zoom}.{curr_tile_x}")
```

Comment:
Why "Interval 0"?

Reply:
Oh I have to remove this. This was only for testing. Good catch!